File: State/cpp/Machine.cpp
#include "Machine.h"
#include "State.h"
#include "OFF.h"

Machine::Machine(){
	current = new OFF();
	cout << '\n';
}

void Machine::setCurrent(State *s){
	current = s;
}

void Machine::on(){
	current->on(this);
}

void Machine::off(){
	Machine::current->off(this);
}

File: State/cpp/java/src/dp/state/sm/Machine.java
package dp.state.sm;

public class Machine {
	State current;
	
	// Constructor
	public Machine(){
		current = new OFF();
		System.out.println();
	}
	
	public void setCurrent(State s){
		current = s;
	}
	
	public void on(){
		current.on(this);
	}
	
	public void off(){
		current.off(this);
	}
}